pkg/xds/envoy/listeners/fault_injection_configurer.go

```diff
@@ -0,0 +1,129 @@
+package listeners
+
+import (
+	envoy_listener "github.com/envoyproxy/go-control-plane/envoy/api/v2/listener"
+	envoy_api_v2_route "github.com/envoyproxy/go-control-plane/envoy/api/v2/route"
+	envoy_filter_fault "github.com/envoyproxy/go-control-plane/envoy/config/filter/fault/v2"
+	envoy_http_fault "github.com/envoyproxy/go-control-plane/envoy/config/filter/http/fault/v2"
+	envoy_hcm "github.com/envoyproxy/go-control-plane/envoy/config/filter/network/http_connection_manager/v2"
+	envoy_type_matcher "github.com/envoyproxy/go-control-plane/envoy/type/matcher"
+	envoy_wellknown "github.com/envoyproxy/go-control-plane/pkg/wellknown"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/wrappers"
+
+	mesh_proto "github.com/Kong/kuma/api/mesh/v1alpha1"
+	"github.com/Kong/kuma/pkg/xds/envoy/routes"
+	"github.com/Kong/kuma/pkg/xds/envoy/tags"
+)
+
+func FaultInjection(faultInjection *mesh_proto.FaultInjection) FilterChainBuilderOpt {
+	return FilterChainBuilderOptFunc(func(config *FilterChainBuilderConfig) {
+		config.Add(&FaultInjectionConfigurer{
+			faultInjection: faultInjection,
+		})
+	})
+}
+
+type FaultInjectionConfigurer struct {
+	faultInjection *mesh_proto.FaultInjection
+}
+
+func (f *FaultInjectionConfigurer) Configure(filterChain *envoy_listener.FilterChain) error {
+	if f.faultInjection == nil {
+		return nil
+	}
+
+	config := &envoy_http_fault.HTTPFault{
+		Delay: convertDelay(f.faultInjection.Conf.GetDelay()),
+		Abort: convertAbort(f.faultInjection.Conf.GetAbort()),
+		Headers: []*envoy_api_v2_route.HeaderMatcher{
+			createHeaders(f.faultInjection.SourceTags()),
+		},
+	}
+
+	rrl, err := convertResponseRateLimit(f.faultInjection.Conf.GetResponseBandwidth())
+	if err != nil {
+		return err
+	}
+	config.ResponseRateLimit = rrl
+
+	pbst, err := ptypes.MarshalAny(config)
+	if err != nil {
+		return err
+	}
+
+	return UpdateHTTPConnectionManager(filterChain, func(manager *envoy_hcm.HttpConnectionManager) error {
+		manager.HttpFilters = append([]*envoy_hcm.HttpFilter{
+			{
+				Name: envoy_wellknown.Fault,
+				ConfigType: &envoy_hcm.HttpFilter_TypedConfig{
+					TypedConfig: pbst,
+				},
+			},
+		}, manager.HttpFilters...)
+		return nil
+	})
+}
+
+func createHeaders(selectors []mesh_proto.SingleValueTagSet) *envoy_api_v2_route.HeaderMatcher {
+	var rss []string
+	for _, selector := range selectors {
+		rs := tags.MatchingRegex(selector)
+		rss = append(rss, rs)
```

Comment:
`rss = append(rss, rs)` what rss and rs stands for? Can we use more descriptive names?